libs/auth-api-lib/src/lib/delegations/admin/delegation-admin-custom.service.ts

```diff
@@ -0,0 +1,133 @@
+
+@Injectable()
+export class DelegationAdminCustomService {
+  constructor(
+    @InjectModel(Delegation)
+    private delegationModel: typeof Delegation,
+    private delegationResourceService: DelegationResourcesService,
+    private delegationIndexService: DelegationsIndexService,
+    private delegationScopeService: DelegationScopeService,
+  ) {}
+
+  async getAllDelegationsByNationalId(
+    nationalId: string,
+  ): Promise<DelegationAdminCustomDto> {
+    const incomingDelegations = await this.delegationModel.findAll({
+      useMaster: true,
+      where: {
+        toNationalId: nationalId,
+      },
+      include: [
+        {
+          model: DelegationScope,
+          required: true,
+          include: [
+            {
+              model: ApiScope,
+              as: 'apiScope',
+              required: true,
+              where: {
+                enabled: true,
+              },
+              include: [
+                {
+                  model: ApiScopeDelegationType,
+                  required: true,
+                  where: {
+                    delegationType: AuthDelegationType.Custom,
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    })
+
+    const outgoingDelegations = await this.delegationModel.findAll({
+      useMaster: true,
+      where: {
+        fromNationalId: nationalId,
+      },
+      include: [
+        {
+          model: DelegationScope,
+          required: true,
+          include: [
+            {
+              model: ApiScope,
+              required: true,
+              as: 'apiScope',
+              where: {
+                enabled: true,
+              },
+              include: [
+                {
+                  model: ApiScopeDelegationType,
+                  required: true,
+                  where: {
+                    delegationType: AuthDelegationType.Custom,
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    })
+
+    return {
+      incoming: incomingDelegations.map((delegation) => delegation.toDTO()),
+      outgoing: outgoingDelegations.map((delegation) => delegation.toDTO()),
+    }
+  }
+
+  async deleteDelegation(user: User, delegationId: string): Promise<void> {
+    // TODO: Check if delegation has a ReferenceId and throw error if it does not.
+    const delegation = await this.delegationModel.findByPk(delegationId)
+
+    if (!delegation) {
+      throw new NotFoundException('Delegation not found')
+    }
+
+    const userScopes = await this.delegationResourceService.findScopes(
+      user,
+      delegation.domainName,
+    )
+
+    await this.delegationScopeService.delete(
+      delegationId,
+      userScopes.map((scope) => scope.name),
+    )
+
+    // If no scopes are left delete the delegation.
+    const remainingScopes = await this.delegationScopeService.findAll(
```

Comment:
Should it be passed as an argument to use master node?

Reply:
I dont think this needs to be passed as as a variable, since `delegationScopeService.findAll`is used in two places and both of them right after a delete.